### ISSUE TITLE
How to forbid checkpoints saving after each epoch?

### ISSUE THREAD

**Issue by RunfengZhang** [facebookresearch/PyTorch-BigGraph#222]
Hi, I found the checkpoints saving is really time-consuming. So I want to save checkpoints after every N epochs' training. Could I realize this just through the change of parameters? Or, should I change the code? Thx!

**Comment by lw** [comment]
I don't think we offer any option to do that out-of-the-box. My feeling is that if checkpointing is such a high fraction of your training time then something might be off (is your model too sparse? are you using very few negatives?). If you really want to do it, you can certainly modify the code (I think [this](https://github.com/facebookresearch/PyTorch-BigGraph/blob/24bd54e603152efaf30613f0a2f6c75e9bdeb1b6/torchbiggraph/train_cpu.py#L708-L710) would be a good point to add a hook), or you could "work around it" for example by using an edge list that is obtained by replicating K times your original edge list, so that each new epoch corresponds to K old epochs (although it's not exactly equivalent).
